packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -84,71 +128,48 @@ contract FundingRateStore is FundingRateStoreInterface, Testable, Lockable {
      */
     function getFundingRateForContract(address perpetual)
```

Comment:
Should we make this method `view` only, and therefore not disperse rewards?

Pro:
- Every perpetual contract method calls `getFundingRateForContract` so it would reduce Perpetual function gas 

Cons:
- Perpetual contract method calls would no longer cause proposal rewards to get dispersed, instead only `propose`, `withdrawProposalReward`, `dispute`, `settleDispute` on the FundingRateStore would disperse rewards. This opens up room for a proposer to "time" their proposal rewards to when the Perpetual's `pfc` is highest (recall that proposal reward is a stored % of the current `pfc`)

Reply:
nvm. I see this is exactly how you deal with it in the tests here: https://github.com/UMAprotocol/protocol/pull/2168/files#r521478112

I'm ok with it being not viewed as we can still compute the output of the function

Reply:
I'm going to leave it as non-view for now, but we should revisit it and highlight to OZ because `getFundingRateForContract` does get called on every single Perpetual method so it contributes non-trivially to gas

Reply:
SGTM